src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,56 @@
+.calendar {
+ 
+  &__month {
+    display: flex;
+    flex-wrap: wrap;
+    gap: $gap-size;
+    padding: $padding;
+    width: width-size($cell-size, $gap-size, $padding);
+  
+    $days:
+      mon 0,
+      tue 1,
+      wed 2,
+      thu 3,
+      fri 4,
+      sat 5,
+      sun 6;
+  
+    @each $name, $day in $days {
+      &--start-day_#{$name} .calendar__day:first-child {
+        margin-left: margin-size($cell-size, $gap-size, $day);
+      }
+    }
+
+    @for $length from 28 through 31 {
+      &--month-length_#{$length} .calendar__day:nth-child(#{$length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    @include square($cell-size);
+  
+    border: solid 1px black;
+    background-color: #eee;
+    text-align: center;
+    line-height: 98px;
+  
+    @for $number from 1 through 31 {
+      &:nth-child(#{$number})::before {
+        content: "#{$number}";
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translate(0, -20px);
+      transition: 0.5s;
```

Comment:
Move the 'transition' property to the '&__day' selector, so that it affects both the hover-in and hover-out animations.